full/Dockerfile

```diff
@@ -1,192 +1,153 @@
-ARG SHA=ce50b1c91364cb77efe3776f756a6d92b76d9038b0a0782f7d53acf1e997a14d
-ARG BASE_URL=https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries
-
-RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
-  && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
-  && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c - \
-  && tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1 \
-  && rm -f /tmp/apache-maven.tar.gz \
-  && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
-ENV MAVEN_HOME /usr/share/maven
-ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
-ENV PATH $PATH:${MAVEN_HOME}/bin
-
-#C/C++
-RUN curl https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-RUN echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-6.0 main" > /etc/apt/sources.list.d/llvm.list
-RUN apt-get update \
-    && apt-get install -y \
-        clang-tools-6.0 \
-        cmake \
-        clang-format \
-    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
-RUN ln -s /usr/bin/clangd-6.0 /usr/bin/clangd
-
-# PHP
-ENV DEBIAN_FRONTEND=noninteractive
-RUN apt-get update \
-    && apt-get install -y \
-        php \
-        apache2 \
+ENV MAVEN_HOME=/usr/share/maven \
+    MAVEN_CONFIG=$HOME/.m2
+ENV PATH=$MAVEN_HOME/bin:$PATH
+RUN mkdir -p $MAVEN_HOME \
+    && curl -fsSL https://apache.osuosl.org/maven/maven-3/$MAVEN_VERSION/binaries/apache-maven-$MAVEN_VERSION-bin.tar.gz \
+        | tar -xzvC $MAVEN_HOME --strip-components=1
+
+### PHP ###
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -yq \
         composer \
+        php \
         php-all-dev \
-        php-intl \
-        php-mysql \
-        php-pgsql \
-        php-xml \
+        php-ctype \
+        php-curl \
+        php-date \
         php-gd \
+        php-gettext \
+        php-intl \
         php-json \
-        php-curl \
         php-mbstring \
+        php-mysql \
+        php-net-ftp \
+        php-pgsql \
         php-sqlite3 \
         php-tokenizer \
-        php-ctype \
-        php-date \
-        php-net-ftp \
-        php-gettext \
+        php-xml \
         php-zip \
-    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
- # the PHP language server is installed by the theia-php-extension
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/*
+# PHP language server is installed by theia-php-extension
+
+### Yarn ###
+RUN curl -fsSL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && apt-add-repository -yu "deb https://dl.yarnpkg.com/debian/ stable main" \
+    && apt-get install --no-install-recommends -yq yarn=1.12.3-1 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/*
+
+### Gitpod user (2) ###
+USER gitpod
+# use sudo so that user does not get sudo usage info on (the first) login
+RUN sudo echo "Running 'sudo' for Gitpod: success"
+
+### Go ###
+ENV GO_VERSION=1.11.2 \
```

Comment:
should the GO_VERSION be an ARG too?

Reply:
maybe. I'd have to check all the languages how they should be installed, what should be all the time in the environment etc. I tried to touch these things only when I saw obvious duplication or had an issue with that ...

Reply:
I'm fine with leaving it as is.

Reply:
in the end I checked in detail Python because that I touched and Alex was unsure whether it's OK ... so, it should be OK :-)